notebooks/en/annotate_text_data_transformers_via_active_learning.ipynb

```diff
@@ -85,7 +85,7 @@
       },
```

Comment:
<p>why not load_dataset though, is there an issue?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/139/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
I have the same concerns here.

I've tried two approaches:

* Using the datasets package versions shown in the notebook (2.9.0), keeping the load_dataset `X_labeled_full = load_dataset("Cleanlab/stanford-politeness", split="labeled")` and downgrading Pandas to 1.5.0. This way, it only detects the 'test' split. `ValueError: Unknown split "labeled". Should be one of ['test'].`. Additionally, it generates and error in the following cell `AttributeError: 'Dataset' object has no attribute 'head'`
* Using the upgraded dataset package version (2.20.0) and load_dataset `X_labeled_full = load_dataset('csv', data_files="https://raw.githubusercontent.com/cmauck10/data/main/stanford-politeness/X_labeled_full.csv")`. In the following cell, it generates an error `AttributeError: 'DatasetDict' object has no attribute 'head'` because it expects a Pandas DataFrame.

What do you think? 🤔

Reply:
I think you need to select the split of the dataset, i.e. `ds = load_dataset(..)["train"]` and then wrap with `pd.DataFrame` if you really want to convert to pandas. `DatasetDict` often has the splits as a dict and one needs to select which dataset they want to modify etc.

Reply:
The `load_dataset()` does not seem to work with the current code.

The notebook is a "copy" of the one on this web:  https://cleanlab.ai/blog/active-learning-transformers/ (inside it says Colab notebook where the notebook link is: https://colab.research.google.com/github/cleanlab/examples/blob/master/active_learning_transformers/active_learning.ipynb).

The main difference with that notebook is the use of `load_dataset` with the dataset in HF datasets. Running the code using any of the following:

* `X_labeled_full = load_dataset("Cleanlab/stanford-politeness")["labeled"]`
* `X_labeled_full = load_dataset("Cleanlab/stanford-politeness", split="labeled")`
* `dataset = load_dataset('Cleanlab/stanford-politeness')`

produces the following error:

````
---------------------------------------------------------------------------
CastError                                 Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/datasets/builder.py](https://localhost:8080/#) in _prepare_split_single(self, gen_kwargs, fpath, file_format, max_shard_size, job_id)
   2012                     try:
-> 2013                         writer.write_table(table)
   2014                     except CastError as cast_error:

8 frames
CastError: Couldn't cast
id: string
text: string
-- schema metadata --
pandas: '{"index_columns": [{"kind": "range", "name": null, "start": 0, "' + 474
to
{'id': Value(dtype='string', id=None), 'text': Value(dtype='string', id=None), 'a6': Value(dtype='float64', id=None), 'a12': Value(dtype='float64', id=None), 'a16': Value(dtype='float64', id=None), 'a19': Value(dtype='float64', id=None), 'a20': Value(dtype='float64', id=None), 'a22': Value(dtype='float64', id=None), 'a39': Value(dtype='float64', id=None), 'a42': Value(dtype='float64', id=None), 'a52': Value(dtype='float64', id=None), 'a60': Value(dtype='float64', id=None), 'a61': Value(dtype='float64', id=None), 'a65': Value(dtype='float64', id=None), 'a68': Value(dtype='float64', id=None), 'a70': Value(dtype='float64', id=None), 'a93': Value(dtype='float64', id=None), 'a98': Value(dtype='float64', id=None), 'a99': Value(dtype='float64', id=None), 'a100': Value(dtype='float64', id=None), 'a102': Value(dtype='float64', id=None), 'a120': Value(dtype='float64', id=None), 'a127': Value(dtype='float64', id=None), 'a131': Value(dtype='float64', id=None), 'a151': Value(dtype='float64', id=None), 'a157': Value(dtype='float64', id=None), 'a158': Value(dtype='float64', id=None), 'a178': Value(dtype='float64', id=None), 'a180': Value(dtype='float64', id=None), 'a185': Value(dtype='float64', id=None), 'a193': Value(dtype='float64', id=None), 'a196': Value(dtype='float64', id=None), 'a197': Value(dtype='float64', id=None), 'a215': Value(dtype='float64', id=None), 'a216': Value(dtype='float64', id=None)}
because column names don't match

During handling of the above exception, another exception occurred:

DatasetGenerationCastError                Traceback (most recent call last)
[/usr/local/lib/python3.10/dist-packages/datasets/builder.py](https://localhost:8080/#) in _prepare_split_single(self, gen_kwargs, fpath, file_format, max_shard_size, job_id)
   2013                         writer.write_table(table)
   2014                     except CastError as cast_error:
-> 2015                         raise DatasetGenerationCastError.from_cast_error(
   2016                             cast_error=cast_error,
   2017                             builder_name=self.info.builder_name,

DatasetGenerationCastError: An error occurred while generating the dataset

All the data files must have the same columns, but at some point there are 33 missing columns ({'a61', 'a93', 'a196', 'a12', 'a185', 'a178', 'a16', 'a180', 'a19', 'a127', 'a102', 'a98', 'a99', 'a157', 'a68', 'a197', 'a52', 'a120', 'a215', 'a193', 'a131', 'a20', 'a151', 'a100', 'a65', 'a216', 'a42', 'a158', 'a22', 'a70', 'a39', 'a6', 'a60'})

This happened while the csv dataset builder was generating data using

hf://datasets/Cleanlab/stanford-politeness/X_unlabeled.csv (at revision aa232bcccc928f232c50321ab3d8fffa05414181)

Please either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)
````

It can be converted to `DatasetDict` using:

````
X_labeled_full = load_dataset('csv', data_files="https://raw.githubusercontent.com/cmauck10/data/main/stanford-politeness/X_labeled_full.csv")
````

which is:

````
DatasetDict({
    train: Dataset({
        features: ['id', 'text', 'a6', 'a12', 'a16', 'a19', 'a20', 'a22', 'a39', 'a42', 'a52', 'a60', 'a61', 'a65', 'a68', 'a70', 'a93', 'a98', 'a99', 'a100', 'a102', 'a120', 'a127', 'a131', 'a151', 'a157', 'a158', 'a178', 'a180', 'a185', 'a193', 'a196', 'a197', 'a215', 'a216'],
        num_rows: 100
    })
})
````

but this doesn't work since the rest of the notebook uses pandas instead of HF Dataset or DatasetDict (for example the following cell uses `X_labeled_full.head()`).

I guess that the options are:
* Download the dataset as a pandas dataset directly (as I propose in this PR)
* Rewrite deeply the code to use `Datasets` and `DatasetDict` (ideal solution).

Reply:
Hi @merveenoyan 😄 
I've updated the notebook with a new possible solution that now uses `load_dataset` and `DatasetDict`. I believe this solution is more HF-friendly since it uses the available tools 🤗 Tell me what you think!